### DISCUSSION TITLE
How to set secrets?

### DISCUSSION THREAD

**Discussion by poperigby** [Kozea/Radicale#1564] [Q&A server] [answered]
I'm trying to setup Radicale using the [NixOS module](https://mynixos.com/nixpkgs/options/services.radicale), but I need to give Radicale a password for [this](https://github.com/malmeloo/radicale3-ldap-auth) plugin. I don't want to put it in the configuration file, so is there some way to set option with environment variables or something?

**Comment by pbiering** [comment, marked as answer]
all options in config file can be also set by command line argument, support of fetching data from environment for a particular option is not supported so far.

**Reply by poperigby** [reply]
Got it. Can I use both a configuration file and command line arguments? I want most of my configuration to be in a file, and sensitive values to be passed through arguments.

**Reply by pbiering** [reply]
command line options are honored after the config file is read.

**Comment by pbiering** [comment]
btw: the mentioned plugin is potentially no longer needed as native support is merged now by https://github.com/Kozea/Radicale/pull/1218
